grr/server/grr_response_server/flows/general/osquery_test.py

```diff
@@ -147,48 +152,108 @@ def testFailure(self):
       self._RunQuery("UPDATE time SET day = -1;")
 
 
-class FakeOsqueryFlowTest(flow_test_lib.FlowTestsBaseclass):
+def _CreateDictTable(column_number, row_number) -> List[OrderedDict]:
+  def CellValue(col_number, row_number):
+    return (f"col-{col_number}", f"col-{col_number}, row-{row_number}")
+
+  return [
+    OrderedDict(
+      CellValue(col_number, row_number) for col_number in range(column_number)
+    ) for row_number in range(row_number)
+  ]
 
+
+class FakeOsqueryFlowTest(flow_test_lib.FlowTestsBaseclass):
   def setUp(self):
     super(FakeOsqueryFlowTest, self).setUp()
     self.client_id = self.SetupClient(0)
 
-  def _RunQuery(self, query):
+  def _NewTestFlowStartAndRun(self, query):
     session_id = flow_test_lib.TestFlowHelper(
         osquery_flow.OsqueryFlow.__name__,
         action_mocks.ActionMock(osquery_action.Osquery),
         client_id=self.client_id,
         token=self.token,
         query=query)
+    return session_id
+
+  def _RunQuery(self, query):
+    session_id = self._NewTestFlowStartAndRun(query)
     return flow_test_lib.GetFlowResults(self.client_id, session_id)
 
+  def _AssertTablesMatch(
+    self,
+    table: rdf_osquery.OsqueryTable,
+    expected: List[OrderedDict]
+  ) -> None:
+    first_row = expected[0]
+    expected_columns = list(first_row.keys())
+
+    self.assertLen(table.header.columns, len(expected_columns))
+
+    column_pairs = zip(table.header.columns, expected_columns)
+    for table_col, expected_col_name in column_pairs:
+      self.assertEqual(table_col.name, expected_col_name)
+
+    for col_name in expected_columns:
+      table_col_values = list(table.Column(col_name))
+      expected_col_vlues = [row[col_name] for row in expected]
+
+      self.assertEqual(table_col_values, expected_col_vlues)
+
   def testSuccess(self):
-    stdout = """
-    [
-      { "foo": "quux", "bar": "norf", "baz": "thud" },
-      { "foo": "blargh", "bar": "plugh", "baz": "ztesch" }
+    test_table = [
+      OrderedDict([ ("foo", "quux"), ("bar", "norf"), ("baz", "thud") ]),
+      OrderedDict([ ("foo", "blargh"), ("bar", "plugh"), ("baz", "ztesch") ]),
     ]
-    """
+    stdout = json.Dump(test_table)
```

Comment:
Why? This contains more logic and is more complicated (it has more structure, yes, but this is just test data, do we care?).